Feature/generic type and member

BI003 opened the pull request.

BI003 committed:
В проект BankSystem.App добавлена папка Interfaces, в которой реализован интерфейс IStorage<T> описывающий хранилище.
Интерфейс содержит методы List<T> Get(какой-то фильтр), Add(T item), Update(T item), Delete(T item).
● В папке Interfaces добавлен интерфейс “IClientStorage” описывающий хранилище клиентов. Интерфейс имплементирует “IStorage”, содержит методы - AddAccount, UpdateAccount, DeleteAccount.
● В созданном ранее хранилище клиентов имплементирован интерфейс “IClientStorage ” и реализованы его методы.
● В классе “ClientService”, вместо ранее применяемого типа “ClientStorage” применён “IClientStorage” (поле и конструктор).
● Убедился что проекты в решении ссылаются в следующем порядке Infrastructurе => Application => Domain, и при этом нет конфликтов.

BI003 committed:
Перестроена архитектура работы с сотрудником банка, базируясь на примере с клиентами.

BI003 committed:
В сервисе “BankService” реализованы методы:
а) AddBonus принимает на вход наследника класса Person и добавляет ему бонус;
б) AddToBlackList<T>(T person) - добавляет сущность в список хранящийся в этом же классе;
в) IsPersonInBlackList<T>(T person) - проверяет присутствует ли сущность в черном списке

BI003 committed:
Небольшие доработки и упрощения.

PetrovskiAlex commented on BankSystem.App/Interfaces/IEmployeeStorage.cs:
```diff
@@ -0,0 +1,19 @@
+{
+    public interface IEmployeeStorage : IStorage<Employee>
+    {
+        void AddAccount(int passport, Account account);
```
здесь не нужны были эти методы

PetrovskiAlex commented on BankSystem.App/Interfaces/IEmployeeStorage.cs:
```diff
@@ -0,0 +1,19 @@
+
+        IEnumerable<Employee> GetAllEmployees();
+
+        Employee GetEmployeeByPassport(int passport);
```
Зачем этот метод если есть метод Get принимающий предикат?

PetrovskiAlex commented on BankSystem.App/Services/ClientService.cs:
```diff
@@ -90,12 +98,17 @@ public void AddAdditionalAccount(int passport, Account newAccount)
         public IEnumerable<Client> GetFilteredClients(Func<Client, bool> filter = null)
```
в storage ведь есть метод который принимает предикат, зачем тут снова писать такой метод?

BI003 committed:
Упростил метод GetFilteredClients в классе ClientService, используя метод Get из хранилища напрямую, а не переписывая логику фильтрации.

BI003 committed:
Изменён интерфейс IEmployeeStorage, удалены ненужные методы.
Методы GetEmployeeByPassport заменены на использование метода Get.
Метод GetFilteredEmployee упрощён и использует напрямую метод Get из хранилища.

BI003 committed:
Внёс правки в тесты, в соответствии с изменениями.
